pkg/pillar/cmd/volumemgr/README.md

```diff
@@ -0,0 +1,149 @@
+# Volume Manager
+
+Volume Manager (volumemgr) is responsible for creating volumes and managing the
+immutable binary bits that sit behind them.
+
+This is (incomplete) documentation of how volumemgr works.
+
+## Download and Verify
+
+Every volume is created from bits that represent an image - disk or container -
+although it is possible to create a volume from a blank image.
+
+To get the image, we need to _download_ it and _verify_ it.
+
+* Download: get it from somewhere in the network
+* Verify: check that the image contents match a separately-provided hash
+
+The term "image" is used interchangeably with "content tree". This is because
+an image might be a single binary blob, or it might be a "tree", i.e.
+a root manifest with children blobs, including other manifests and one or more
+binary blobs, as well as instructions how to merge the blobs together to create
+a single image.
+
+### Security
+
+For security reasons, EVE does not allow processes that can talk to the network
+be the same processes that verify the bits that are downloaded. Thus, the
+process is:
+
+1. Instruct a network-connected process to download bits
+1. Instruct a non-network-connected process to verify the bits
+1. Use the bits
+
+### Asynchronous Programming
+
+Volume Manager, like most of the rest of EVE, works using asynchronous
+messages via pubsub. Each task is marked as complete, and a new task is started,
+by creating and passing the right message, which a different process will consume.
+
```

Comment:
These initial 39 lintes needs to be edited into the volumemgr.md so that the flow makes sense.
The rest can presumably be appended to that file.

Reply:
OK